### ISSUE TITLE
scrolling code page is so laggy

### ISSUE THREAD

**Issue by alireza0sfr** [microsoft/vscode#215620]
Type: <b>Bug</b>

Hi, this issue keeps hepenning for me in the latest update
when I scroll on my code page it's not smooth and it's so laggy
and my cpu usage jumps from 5% to 95%

VS Code version: Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:43.666Z)
OS version: Linux x64 5.19.0-46-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-10300H CPU @ 2.50GHz (8 x 3663)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|4, 2, 1|
|Memory (System)|15.45GB (9.89GB free)|
|Process Argv|--force-user-env --no-sandbox --use-gl=angle --use-angle=swiftshader --crash-reporter-id 99925ace-3182-4268-9b5e-c102a8f570c9|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|wayland|
</details><details><summary>Extensions (31)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.5.0
vscode-sqlite|ale|0.14.1
vscode-django|bat|1.15.0
vscode-tailwindcss|bra|0.10.5
npm-intellisense|chr|1.4.5
path-intellisense|chr|2.9.0
cucumber-official|Cuc|1.10.0
githistory|don|0.6.20
gitlens|eam|2024.6.1505
vscode-npm-script|eg2|0.3.29
vsc-material-theme|Equ|34.4.0
search-node-modules|jas|1.3.0
vscode-commitizen|Kni|1.1.0
string-manipulation|mar|0.5.7
vscode-docker|ms-|1.29.1
vscode-postgresql|ms-|0.3.0
autopep8|ms-|2024.0.0
black-formatter|ms-|2024.2.0
debugpy|ms-|2024.6.0
python|ms-|2024.8.1
vscode-pylance|ms-|2024.6.1
vscode-typescript-next|ms-|5.6.20240614
nuxtr-vscode|Nux|0.2.16
docker-compose|p1c|0.5.1
material-icon-theme|PKi|5.4.2
LiveServer|rit|5.7.9
volar|Vue|2.0.10
nuxt-typescript-snippets|wil|0.1.3
JavaScriptSnippets|xab|1.8.0
vscode-python-poetry|zes|0.17.21
material-theme|zhu|3.17.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627cf:30244335
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelc:31048052
dsvsc021:30996838
724cj586:31013169
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupytercf:31046870

```

</details>

**Comment by shelvesdragon** [comment]
This is likely caused by the temporary fix in version 1.90.1 on the snap package that switched the OpenGL backend to software emulation in order to prevent the crashes reported with version 1.90.0 in #212494 . If it is related, your options for now are mentioned in this comment: https://github.com/microsoft/vscode/issues/212494#issuecomment-2168072242
Hope this helps!

**Comment by alireza0sfr** [comment]
I didn't want to use code-insiders and configure it to fit my style with my extenstions
but I reverted the code to this and it's fixed

`sudo snap revert code --revision 160`

Thanks alot
